devices/YarpRobotLoggerDevice/src/YarpRobotLoggerDevice.cpp

```diff
@@ -1366,115 +1507,135 @@ void YarpRobotLoggerDevice::run()
+                             + cartesianWrenchName;
+                logData(signalName, m_ftBuffer, time);
+            }
         }
     }
 
-    std::string signalFullName;
     for (auto& [name, signal] : m_vectorsCollectionSignals)
     {
         std::lock_guard<std::mutex> lock(signal.mutex);
-        const BipedalLocomotion::YarpUtilities::VectorsCollection* collection
+        BipedalLocomotion::YarpUtilities::VectorsCollection* externalSignalCollection
             = signal.client.readData(false);
-        if (collection != nullptr)
+
+        if (externalSignalCollection != nullptr)
         {
-            if (!signal.dataArrived)
+            if(!signal.dataArrived)
             {
-                for (const auto& [key, vector] : collection->vectors)
+                for (const auto& [key, vector] : externalSignalCollection->vectors)
                 {
-                    signalFullName = signal.signalName + "::" + key;
+                    signalName = signal.signalName + treeDelim + key;
                     const auto& metadata = signal.metadata.vectors.find(key);
                     if (metadata == signal.metadata.vectors.cend())
                     {
                         log()->warn("{} Unable to find the metadata for the signal named {}. The "
                                     "default one will be used.",
                                     logPrefix,
-                                    signalFullName);
-                        m_bufferManager.addChannel({signalFullName, {vector.size(), 1}});
+                                    signalName);
+                        initMetadata(signalName, {});
                     } else
                     {
-                        // if the metadata is found we use it
-                        m_bufferManager.addChannel({signalFullName, //
-                                                    {vector.size(), 1},
-                                                    metadata->second});
+                        initMetadata(signalName, {metadata->second});
                     }
-
-                    signal.dataArrived = true;
                 }
+                signal.dataArrived = true;
             }
-
-            for (const auto& [key, vector] : collection->vectors)
+            else
             {
-                signalFullName = signal.signalName + "::" + key;
-                m_bufferManager.push_back(vector, time, signalFullName);
+                for (const auto& [key, vector] : externalSignalCollection->vectors)
+                {
+                    signalName = signal.signalName + treeDelim + key;
+                    m_bufferManager.push_back(vector, time, signalName);
+                    if(m_sendDataRT)
+                    {
+                        rtSignalName
+                            = robotRtRootName + treeDelim + signalName;
+                        m_vectorCollectionRTDataServer.populateData(rtSignalName, vector);
+                    }
```

Comment:
Why this does not use ``logData``?

Reply:
It should, I missed this one. Fixed in [8f0e4a8](https://github.com/ami-iit/bipedal-locomotion-framework/commit/8f0e4a81bc093986a4829600563a04437575a001)